gitlab/tool.gpt

```diff
@@ -0,0 +1,227 @@
+Share Context: GitLab Context
+Credential: ./credential
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool listMyAssignedMergeRequests
+
+---
+Name: List My Opened Merge Requests
+Description: List GitLab Merge Requests opened by the current user.
+Share Context: GitLab Context
+Credential: ./credential
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool listMyOpenedMergeRequests
+
+---
+Name: Get Merge Request Details
+Description: Show detailed information about a single Merge Request.
+Share Context: GitLab Context
+Credential: ./credential
+Param: issue_iid: The numeric internal ID of a Merge Request.
+Param: project_id: The numeric ID or name of a project in the format `namespace/project_name`.
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool getMergeRequestDetails
+
+---
+Name: Create Merge Request
+Description: Create a new Merge Request.
+Share Context: GitLab Context
+Credential: ./credential
+Param: project_id: The numeric ID or name of a project in the format `namespace/project_name`.
+Param: mr_name: The new Merge Request title.
+Param: mr_description: The new Merge Request description.
+Param: mr_source_branch: The source git branch.
+Param: mr_target_branch: The target git branch.
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool createMergeRequest
+
+---
+Name: Update Merge Request
+Description: Update values in an existing Merge Request.
+Share Context: GitLab Context
+Credential: ./credential
+Param: issue_iid: The numeric internal ID of a Merge Request.
+Param: project_id: The numeric ID or name of a project in the format `namespace/project_name`.
+Param: updates: A JSON-formatted string that conforms to the GitLab API, containing fields to be updated on a Merge Request.
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool updateMergeRequest
+
+---
+Name: Delete Merge Request
+Description: Deletes a Merge Request.
+Share Context: GitLab Context
+Credential: ./credential
+Param: mr_id: The numeric internal ID of a Merge Request.
+Param: project_id: The numeric ID or name of a project in the format `namespace/project_name`.
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool deleteMergeRequest
+
+---
+Name: Add Merge Request Comment
+Description: Adds a new comment on an existing Merge Request.
+Share Context: GitLab Context
+Credential: ./credential
+Param: mr_id: The numeric internal ID of a Merge Request.
+Param: project_id: The numeric ID or name of a project in the format `namespace/project_name`.
+Param: comment: The text to use as the body of the comment. Can contain markdown.
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool addMergeRequestComment
+
+---
+Name: Approve Merge Request
+Description: Approves a Merge Request.
+Share Context: GitLab Context
+Credential: ./credential
+Param: mr_id: The numeric internal ID of a Merge Request.
+Param: project_id: The numeric ID or url-escaped name of a Project.
+
+#!${GPTSCRIPT_TOOL_DIR}/bin/gptscript-go-tool approveMergeRequest
+
+---
+Name: Lookup User ID
```

Comment:
nit: may also be nice to have a `Search Users` tool.

Reply:
Perhaps, but it was not part of the scope in the original issue. `Lookup User ID` exists only to get the user ID for calls that require a numeric ID instead of accepting a username.